rudder-core/src/main/scala/com/normation/rudder/domain/nodes/Node.scala

```diff
@@ -135,22 +180,41 @@ object CompareProperties {
    *   the property
    * - if the value is the emtpy string, remove
    *   the property
+   *
+   * Each time, we have to check the provider of the update to see if it's compatible.
+   * Node that in read-write mode, the provider is the last who wrote the property.
    */
-  def updateProperties(props: Seq[NodeProperty], updates: Option[Seq[NodeProperty]]) = {
+  def updateProperties(props: Seq[NodeProperty], updates: Option[Seq[NodeProperty]], provider: NodePropertyProvider): Box[Seq[NodeProperty]] = {
+    import NodePropertyMode._
     updates match {
-      case None => props
+      case None => Full(props)
       case Some(u) =>
-        val values = u.map { case NodeProperty(k, v) => (k, v)}.toMap
+        val values = u.map { case NodeProperty(k, v, p, m) => (k, (v, p, m))}.toMap
         val existings = props.map(_.name).toSet
-        //for news values, don't keep empty
-        val news = (values -- existings).collect { case(k,v) if(v != JString("")) => NodeProperty(k,v) }
-        props.flatMap { case p@NodeProperty(name, value)  =>
-          values.get(name) match {
-            case None              => Some(p)
-            case Some(JString("")) => None
-            case Some(x)           => Some(NodeProperty(name, x))
-          }
-        } ++ news
+        //for news values, don't keep empty, that make no sense
+        val news = (values -- existings).collect { case(k, (v, p, m)) if(v != JString("")) => NodeProperty(k,v,p,m) }
+
+        //for old value, update only according to rights
+        for {
+          updated <- sequence(props) { case prop@NodeProperty(name, value, provider, mode)  =>
+                       values.get(name) match {
+                         case None            => Full(Some(prop))
+                         case Some((v, p, m)) =>
+                           (mode, provider, p) match {
```

Comment:
Do we need to match on provider and p ? 

I think you only need to match on 'mode' and not on providers ? 

(first case will be 

case Some(ReadOnly)  if provider == p